File: solutions/1. Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];
        HashMap<Integer, Integer> hm = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if(hm.containsKey(diff)){
                res[0] = i;
                res[1] = hm.get(diff);
                return res;
            }
            hm.put(nums[i], i);
        }
        return res;
    }
}

File: solutions/1. Two Sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    hash = {}
    nums.each_with_index do |num, index|
        if hash[target - num]
            return [hash[target - num], index]
        else
            hash[num] = index
        end
    end
end
​
